### PR TITLE
Add support for zoom

### PR DESCRIPTION
**Pull request by dvkch**
Hi again,

I am currently working on turning an old video conferencing camera into a ONVIF compatible camera. Since the original hardware supports optical zooming I have added the required elements to ONVIF simple server to support those.

I tried to check with the T profile specifications carefully when adding this support, and made sure not to introduce regressions in the existing code for projects that don't use zoom. I hope I have done well on both accounts, but feel free to ask for any changes.

I have also added support for (optional) minimum values for x, y and z in the configuration file, defaulting to 0 if missing. My camera doesn't rotate completely, I ended up putting its coordinate system origin to middle pan, middle tilt and minimum zoom. The only part where I was unsure of their use was for TranslationSpaceFov handling I left it as is for now. If you don't want to support this I can also remove this part from the PR.

Thanks for considering :)

### CODE REVIEW

**Comment by roleoroleo** [comment]
Thank you for your PR.

I ask you if you can make some small fix.
1 - Could you check the cat function? The 3rd parameter is the number of following strings; when you add a new string to replace you have to change this number. For example line 145 of ptz_service.c should be 12.
2 - Line 499, 507, etc.., I don't understand the new parameter "dx" you added to move_* commands. The function continuous_move doesn't use a quantity because it just starts the movement (doesnt' stop it). Please explain me better.

About TranslationSpaceFov I don't know exactly how the zoom should be changed inside this command. Maybe your implementation is ok.

**Comment by dvkch** [comment]
1/ I’ll make those changes as soon as I’m on my computer. I hadn’t realized that! 

2/ the parameter is actually the speed requested by the client. It is between -1 and 1, the app IPCams in the AppStore uses those to send continuous move requests with different speeds. I saw that you already parsed them properly, simply giving them to the continuous move commands allowed me to map it to my camera ptz requests and have it all work properly. 

3/ I’ll look more into the specs to know if there can be a zoom argument for the Fov, and if so try to support it properly or with reasonable assumptions.

**Comment by dvkch** [comment]
I have made updates to fix the `cat` issue, as well as rebased on your current master since the media files refacto was conflicting with this patch.

As per Fov, zoom isn't supposed to be included in those requests, the spec v1706 only mentions Pan and tilt translation spaces for this request, as such I have removed it from relative move arguments in this context

**Comment by roleoroleo** [comment]
Thank you.
I will work on your pr the next weekend.

File: ptz_service.c [hunk 1]
```diff
@@ -128,128 +128,188 @@ int ptz_get_service_capabilities()
...
-    long size = cat(NULL, "ptz_service_files/GetConfiguration.xml", 8,
-            "%MIN_X%", "0.0",
+    long size = cat(NULL, "ptz_service_files/GetConfiguration.xml", 12,
+            "%MIN_X%", min_x,
             "%MAX_X%", max_x,
-            "%MIN_Y%", "0.0",
-            "%MAX_Y%", max_y);
+            "%MIN_Y%", min_y,
+            "%MAX_Y%", max_y,
+            "%MIN_Z%", min_x,
```
Comment (roleoroleo) [inline ptz_service.c:186]: It should be min_z

File: ptz_service.c [hunk 2]
```diff
@@ -128,128 +128,188 @@ int ptz_get_service_capabilities()
...
-    long size = cat(NULL, "ptz_service_files/GetNode.xml", 8,
-            "%MIN_X%", "0.0",
+    long size = cat(NULL, "ptz_service_files/GetNode.xml", 12,
+            "%MIN_X%", min_x,
             "%MAX_X%", max_x,
-            "%MIN_Y%", "0.0",
-            "%MAX_Y%", max_y);
+            "%MIN_Y%", min_y,
+            "%MAX_Y%", max_y,
+            "%MIN_Y%", min_z,
```
Comment (roleoroleo) [inline ptz_service.c:300]: %MIN_Z%

File: ptz_service.c [hunk 3]
```diff
@@ -128,128 +128,188 @@ int ptz_get_service_capabilities()
...
-    return cat("stdout", "ptz_service_files/GetNode.xml", 8,
-            "%MIN_X%", "0.0",
+    return cat("stdout", "ptz_service_files/GetNode.xml", 12,
+            "%MIN_X%", min_x,
             "%MAX_X%", max_x,
-            "%MIN_Y%", "0.0",
-            "%MAX_Y%", max_y);
+            "%MIN_Y%", min_y,
+            "%MAX_Y%", max_y,
+            "%MIN_Y%", min_z,
```
Comment (roleoroleo) [inline ptz_service.c:311]: %MIN_Z%

**Review by roleoroleo — CHANGES_REQUESTED** [review]
There are a couple of typos. Could you fix them?

**Comment by dvkch** [comment]
Sorry about those typos, these should be fixed now

**Pull request merged**